File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        length = len(prices)

        for i in range(length):
            for j in range(i, length):
                diff = prices[j] - prices[i]
                if diff > max_profit:
                    max_profit = diff

        return max_profit
        

File: Data Structures & Algorithms/two-integer-sum/submission-3.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map: HashMap<i32,i32> = HashMap::new();
        let len = nums.len();

        for i in 0..len {
            let diff = target - nums[i];

            if let Some(val) = map.get(&diff) {
                return vec![*val, i as i32];
            }
            map.insert(nums[i], i as i32);
        }

        vec![]
    }
}
